src/include/settings/settings_defs.h

```diff
@@ -413,6 +413,15 @@ SETTING_bool(
     noisepage::settings::Callbacks::CompiledQueryExecution
 )
 
+SETTING_bool(
+    clear_query_cache,
```

Comment:
I feel we are also generally trending towards adding a lot of self-driving "settings" that should really be functions

Reply:
An option is to invest the work now in implementing just one bytecode, `self_driving_action(action_enum)`
To implement all of the self driving functions based on the `action_enum` value passed in?

Reply:
We'll punt on this and I've created an issue to track this (#1622).